### PR TITLE
Order Creation: Products section UI

### PR DESCRIPTION
**Pull request by rachelmcr**
Part of: #5408

## Description

This adds the Products section UI to the New Order screen in the order creation flow. This section includes:

* A row showing product details and a custom stepper to change the product quantity
* A button to add a new product

For now, the UI is static with temporary, fake data. (Future PRs will add real data and interactions.)

## Changes

* Adds the minus gridicon to `UIImage+Woo`.
* To the `NewOrder` view:
   * Adds a `GeometryReader` to calculate the right edge insets for edge-to-edge backgrounds in landscape.
   * Adds a Products section to the view, including a heading, product row, and add product button.
   * Adds a ProductRow view with a custom stepper for product quantity, to go in the Products section. The custom stepper uses a view preference key to capture the text size, and adjust the stepper layout (button size, padding, spacing) according to that size. This ensures the stepper resizes nicely with dynamic font sizes.
   * Adds a reusable `AddButton` view, since we'll have many similar buttons throughout this screen.

For now, I added these views all within `NewOrder.swift`, but it may make sense to eventually separate them to reuse across different screens.

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button. (If Simple Payments is also enabled, tap "Create order" in the bottom sheet that appears.)
4. On the New Order screen, confirm you see the expected Products section with a product row and `Add product` button.

Accessibility testing:

* Rotate to landscape mode and confirm the layout has the expected padding.
* Change the text size and confirm the layout responds appropriately.
* Build and run the app on device and confirm the VoiceOver controls work well. (Note: For now, the product quantity stepper is using static data so the quantity won't change if you swipe to increment/decrement.)

## Screenshots

Portrait|Landscape
-|-
Simulator Screen Shot - iPhone 13 Pro - 2021-11-25 at 16 33 58|Simulator Screen Shot - iPhone 13 Pro - 2021-11-25 at 16 34 00

Large font size|Dark mode|Right-to-left
-|-|-
NewOrder-Products-Accessibility|NewOrder-Products-Dark|NewOrder-Products-RTL

## Submitter Checklist

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift [hunk 1]
```diff
@@ -45,11 +45,17 @@ struct NewOrder: View {
     @ObservedObject var viewModel: NewOrderViewModel
 
     var body: some View {
-        ScrollView {
-            EmptyView()
-        }
-        .background(Color(.listBackground))
+        GeometryReader { geometry in
```
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift:48]: Apparently there is a new `safeAreaInset()` modifier to help with this use case but it's only available on iOS15 😢

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift [hunk 2]
```diff
@@ -70,11 +76,182 @@ struct NewOrder: View {
...
+    ///
+    struct ProductRow: View {
+        var body: some View {
+            AdaptiveStack(horizontalAlignment: .leading) {
+                HStack(alignment: .top) {
+                    // Product image
+                    // TODO: Display actual product image when available
+                    Image(uiImage: .productPlaceholderImage)
+                        .aspectRatio(contentMode: .fill)
+                        .frame(width: NewOrder.Layout.productImageSize, height: NewOrder.Layout.productImageSize)
```
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift]: Not a big deal, but it would be good if the product image size could also scale with the font size. 
You can do that by using the `scaled metric` property wrapper.

https://github.com/woocommerce/woocommerce-ios/blob/4f8f59560ac35be395e3b9f30782f93a7935b9ef/WooCommerce/Classes/ViewRelated/Orders/Order%20Details/Address%20Edit/FilterListSelector.swift#L65

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift [hunk 3]
```diff
@@ -70,11 +76,182 @@ struct NewOrder: View {
...
+                Button {
+                    // TODO: Decrement the product quantity
+                } label: {
+                    Image(uiImage: .minusImage)
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .frame(height: textSize.height)
+                }
+                Text("1") // Fake data - quantity
+                    .background(ViewGeometry())
```
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift]: Nice, I have never used `PreferenceKeys` before. I did something similar in [`scrollVerticallyIfNeeded`](https://github.com/woocommerce/woocommerce-ios/blob/3d48a920d7c26c89f9cf16a1e5b0a4fbe1d38db9/WooCommerce/Classes/View%20Modifiers/View%2BScrollModifiers.swift) but maybe this is the correct way! 

Do you think we should add some more documentation explaining why is this needed?

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift [hunk 4]
```diff
@@ -70,11 +76,182 @@ struct NewOrder: View {
...
+            }
+        }
+    }
+}
+
+// MARK: Custom Views
+/// Represents a button with a plus icon.
+/// Used for any button that adds details to the order.
+///
+private struct AddButton: View {
```
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift]: As you say, we will be reusing these buttons more and more, I think it makes sense to move this to a `ButtonStyle` so it can be used throughout the app. What do you think?

**Review by Ecarrion — APPROVED** [review]
This looks great @rachelmcr. I learned a lot by renewing this PR 🤓 
Also, loving all the effort added into accessibility ♿ 

I left you some suggestions but nothing blocking you from merging this.

>  but it may make sense to eventually separate them to reuse across different screens.

I think so too!

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift (hunk 1 above)
Comment (rachelmcr) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift:48]: TIL! Too bad it's only iOS15.

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift (hunk 2 above)
Comment (rachelmcr) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift]: TIL about the `scaled metric` property wrapper. That's perfect, thanks!

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift (hunk 3 above)
Comment (rachelmcr) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift]: Yeah, I learned there's a lot that can be done with `PreferenceKeys`! That said, conceptually what I really want to do here is mainly just scale everything as the text size changes. I wasn't aware that I could use `@ScaledMetric` to do that. The only real difference is how I set the initial size of everything (based on the text size or based on a constant).

I think rather than using `PreferenceKeys` in this case, and trying to document it to explain it further, I'll switch to using a scaled metric for sizing here.

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift (hunk 2 above)
Comment (rachelmcr) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift]: Added this resizing in c70545f.

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift (hunk 3 above)
Comment (rachelmcr) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift]: I tried it out and the `scaled metric` approach seems clearer to me; switched to that in bb33741.

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift (hunk 4 above)
Comment (rachelmcr) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift]: Good call. Added a `PlusButtonStyle` button style in 0252eea.
